identity_presentation/src/commonMain/kotlin/net/thechance/mena/identity/presentation/screen/addresses/AddEditLocationScreenViewModel.kt

```diff
@@ -0,0 +1,138 @@
+package net.thechance.mena.identity.presentation.screen.addresses
+
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.IO
+import net.thechance.mena.identity.domain.entity.Address
+import net.thechance.mena.identity.domain.repository.AddressesRepository
+import net.thechance.mena.identity.presentation.base.BaseScreenModel
+import net.thechance.mena.identity.presentation.base.ErrorState
+import net.thechance.mena.identity.presentation.mapper.mapErrorToMessage
+
+class AddEditLocationScreenViewModel(
+    private val addressesRepository: AddressesRepository,
+    private val dispatcher: CoroutineDispatcher = Dispatchers.IO
+) : BaseScreenModel<AddLocationScreenUIState, AddEditLocationScreenUIEffect>(
+    AddLocationScreenUIState()
+), AddEditLocationScreenInteractionListener {
+
+    override fun onClickMap() {
+        sendNewEffect(AddEditLocationScreenUIEffect.NavigateToMap)
+    }
+
+    override fun onClickEdit() {
+        sendNewEffect(AddEditLocationScreenUIEffect.NavigateToMap)
+
+    }
+
+    override fun onClickBack() {
+        sendNewEffect(AddEditLocationScreenUIEffect.NavigateBack)
+
+    }
+
+    override fun onClickAddressType(addressType: AddressType) {
+
+        if (addressType == state.value.addressType) return
+
+        updateState { copy(addressType = addressType) }
+
+        changeIsSaveEnabled()
+
+    }
+
+    override fun onClickSave() {
+        updateState { copy(isLoading = true, errorMessage = null) }
+        tryToExecute(
+            function = {
```

Comment:
I suggest you can extract this to a new function:
```suggestion
            function = ::onSave
```